types/listr/index.d.ts

```diff
@@ -11,41 +12,77 @@ import * as stream from "stream";
 import { Observable } from "rxjs";
 
 declare namespace Listr {
-    interface ListrRenderer {
-        nonTTY: boolean;
-        render(): void;
-        end(): void;
-    }
+    type ListrRendererValue = "silent" | "default" | "verbose" | ListrRendererClass;
 
     interface ListrOptions {
         concurrent?: boolean | number;
         exitOnError?: boolean;
-        renderer?: "silent" | "default" | "verbose" | ListrRenderer;
-        nonTTYRenderer?: "silent" | "default" | "verbose" | ListrRenderer;
+        renderer?: ListrRendererValue;
+        nonTTYRenderer?: ListrRendererValue;
+    }
+
+    type ListrContext = any;
+
+    interface ListrEvent {
+        type: string;
     }
 
+    type ListrTaskResult = string | Promise<any> | Listr | stream.Readable | Observable<any>;
+
     interface ListrTask {
+        title: string;
+        task: (ctx: ListrContext, task: ListrTaskWrapper) => void | ListrTaskResult;
+        skip?: (ctx: ListrContext) => void | boolean | string | Promise<boolean>;
+        enabled?: (ctx: ListrContext) => boolean | Promise<boolean> | Observable<boolean>;
+    }
+
+    interface ListrTaskObject extends Observable<ListrEvent> {
```

Comment:
Add generic, for example:
ListrTaskObject<T = ListrContext > Observable<ListrEvent> {
task: (ctx: T, task: ListrTaskWrapper) => void | ListrTaskResult;